lib/vector-core/src/stream/partitioned_batcher.rs

```diff
@@ -328,6 +331,7 @@ where
             timer: ExpirationQueue::new(settings.timeout),
             partitioner,
             stream: stream.fuse(),
```

Comment:
I think that this `fuse` call here accomplishes the same thing as this PR. See the [docs](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.fuse) for the method.

Reply:
oh nice, you're right!